core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -2055,6 +2055,14 @@ object ZLayer {
       self >>> ZLayer.fromFunctionMany(_.update[A](f))
   }
 
+  implicit final class ZLayerHasROutOps1[RIn, E, A](private val self: ZLayer[RIn, E, Has[A]]) extends AnyVal {
+    /**
+     * Maps layer output which is wrapped in `Has`
+     */
+    def mapDependency[B: Tagged](f: A => B)(implicit A: Tagged[A]): ZLayer[RIn, E, Has[B]] =
```

Comment:
We could potentially call this `mapService`. The issue here is that this only works for layers that output a single service. If we try to call it on layers that output more than one service we can only map over the "first" service and silently discard the rest, which violates the principle that the order of services within `Has` doesn't matter.

Reply:
I see, makes sense. I'll dig into what can be done here to support multiple services, thanks for a good point!

Reply:
Definitely. I think it is going to be very challenging to do this. We would need to define a method on `Has` that would take an existing service `A`, remove it from the map, and add a new service `B`. We can definitely add and remove services from the map, but I think it may not be possible in Scala's type system to describe the type of an intersection with one element removed.

Reply:
Right, just bumped into that. There's another option I'm exploring now - restrict this function to only be used with a single service layer, so that nothing is ever silently lost.

Reply:
So it turned out as quite challenging to implement it in an ergonomic way :)
The best I could do so far is to make it look like this:

```scala
val layer = ???
layer.singleService.map(x => ???)
```
Here `.singleService` will only compile if layer has a single `Has[_]` output.
I think we might want other operations on layers with single outputs in future, so this "narrowing" step might be useful.

Or maybe this is an overkill, WDYT?

Reply:
`RowLacks` is certainly expressible for a finite collection, or with a macro. If Has is to be extended with more operations we'd definitely need it sooner rather than later...

Reply:
@neko-kai I feel this one was intended for some other thread ;)

Reply:
@vpavkin no, this refers to

> but I think it may not be possible in Scala's type system to describe the type of an intersection with one element removed.

Adam's sentence in messages above